src/lib/utils/package-manifest.ts

```diff
@@ -205,10 +247,19 @@ export function getTsEntryPointForPackage(
 ): string | undefined | typeof ignorePackage {
     let packageMain = "index.js"; // The default, per the npm docs.
     let packageTypes = null;
-    if (
+    const typedocPackageConfig = extractTypedocConfigFromPackageManifest(
+        logger,
+        packageJsonPath
+    );
+    if (typedocPackageConfig?.entryPoint) {
+        packageMain = typedocPackageConfig.entryPoint;
+    } else if (
         hasOwnProperty(packageJson, "typedocMain") &&
         typeof packageJson.typedocMain == "string"
     ) {
+        logger.warn(
+            `Legacy typedoc entry point config (using "typedocMain" field) found for "${packageJsonPath}". Please update to use "typedoc": { "entryPoint": "..." } instead. In future upgrade, "typedocMain" field will be ignored.`
```

Comment:
I'd like to pass the path through `nicePath`, which is exported from utils/fs, which will shorten paths if possible.

Reply:
Thanks!